src/Var.jl

```diff
@@ -1078,6 +1087,62 @@ function center_longitude!(var, lon)
     var.data .= shifted_data
 end
 
+"""
+    shift_longitude(var, lower_lon, upper_lon; shift_by = 0.0)
+
+Shift the longitudes in `var` to `lower_lon` to `upper_lon` degrees.
+
+We assume that the units of the longitude dimension is degrees. If this is not the case,
+then one can use [`convert_dim_units`](@ref) to convert the units to degrees.
+
+This function assumes the prime meridian (0th degree) in `var` is the same before and after
+shifting the longitudes.
+
+If two points share the same longitude (e.g. -180 degrees and 180 degrees on longitudes
+spanning from -180 degrees to 180 degrees), then we remove the last longitude before
+shifting the longitudes. This is necessary to prevent duplicated latitudes after shifting
+longitudes.
+
+To shift from -180 to 180 degrees to 0 to 360 degrees, use
+`shift_longitude(var, 0.0, 360.0)` and to shift from 0 to 360 degrees to -180 to 180
+degrees, use `shift_longitude(var, -180.0, 180.0)`.
+"""
+function shift_longitude(var, lower_lon, upper_lon)
+    width = upper_lon - lower_lon
+    (width ≈ 2π) &&
+        @warn "Result may be incorrect if radians are used instead of degrees"
+    (width ≈ 360.0) || error(
+        "The range of longitudes ($lower_lon to $upper_lon degrees) is not 360 degrees",
+    )
+
+    # It is possible for the first and last longitude to represent the same latitude on the
+    # globe. If this is the case, we remove the last point
+    lon_name = longitude_name(var)
+    lon = copy(var.dims[lon_name])
+    if (lon[end] - lon[begin] ≈ 360.0)
+        pop!(lon)
+    end
+
+    # Center longitudes
+    for i in eachindex(lon)
+        if (lon[i] >= upper_lon) || (lon[i] < lower_lon)
+            lon[i] = (lon[i] - lower_lon + width) % width + lower_lon
+        end
+    end
```

Comment:
I wonder if this is more clear:

```suggestion
function wrap_longitude(long, lower, upper, width)
    return (long >= upper || long < lower) ? ((long - lower + width) % width + lower) : long
end

lon .= wrap_longitude.(lon, lower_lon, upper_lon, width)
```

Reply:
In general, could you add two lines of comment to describe the high level approach taken in this function?